brickBreaker.js

```diff
@@ -0,0 +1,217 @@
+  }
+
+  initializeEventListeners() {
+    window.addEventListener("keydown", this.keyDownHandler.bind(this));
+    window.addEventListener("keyup", this.keyUpHandler.bind(this));
+  }
+
+  keyDownHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(-1);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(1);
+        break;
+      case " ":
+        this.reset();
+        break;
+    }
+  }
+
+  keyUpHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(0);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(0);
+        break;
+    }
+  }
+
+  updateEntities() {
+    this.paddle.move();
+    this.ball.move();
+  }
+
+  detectCanvasCollision(ballTop, ballLeft, ballRight) {
+    if (ballLeft <= 0 || ballRight >= this.canvas.width) {
+      this.ball.invertXDirection();
+    } else if (ballTop <= 0) {
+      this.ball.invertYDirection();
+    } else if (ballTop > this.canvas.height) {
+      this.ball.xDirection = 0;
+      this.ball.yDirection = 0;
+    }
+  }
+
+  detectBallCollisionWithEntity(
+    entity,
+    ballTop,
+    ballBottom,
+    ballLeft,
+    ballRight
+  ) {
+    if (entity === this.paddle) {
+      const paddleTop = this.paddle.getTopEdge();
+      const paddleBottom = this.paddle.getBottomEdge();
+      const paddleLeft = this.paddle.getLeftEdge();
+      const paddleRight = this.paddle.getRightEdge();
+
+      if (paddleTop < ballBottom && ballTop >= paddleBottom) {
+        if (paddleLeft <= ballLeft && paddleRight >= ballRight) {
+          this.ball.invertYDirection();
+        }
+      }
+    } else if (entity === this.bricks) {
+      for (let row = 0; row < this.brickRows; row++) {
+        for (let col = 0; col < this.brickCols; col++) {
+          const brick = this.bricks[row][col];
+          const brickTop = brick.getTopEdge();
+          const brickBottom = brick.getBottomEdge();
+          const brickLeft = brick.getLeftEdge();
+          const brickRight = brick.getRightEdge();
+          if (!brick.broken) {
+            if (ballRight >= brickLeft && ballLeft <= brickRight) {
+              if (brickBottom >= ballTop || brickTop >= ballBottom) {
+                this.ball.invertXDirection();
+                this.ball.invertYDirection();
+                brick.broken = true;
+                this.score += 1;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  detectCollisions() {
+    const ballTop = this.ball.getTopEdge();
+    const ballBottom = this.ball.getBottomEdge();
+    const ballLeft = this.ball.getLeftEdge();
+    const ballRight = this.ball.getRightEdge();
+
+    this.detectCanvasCollision(ballTop, ballLeft, ballRight);
+    this.detectBallCollisionWithEntity(
+      this.paddle,
+      ballTop,
+      ballBottom,
+      ballLeft,
+      ballRight
+    );
+    this.detectBallCollisionWithEntity(
+      this.bricks,
```

Comment:
why are you passing in bricks here? you should be detecting collision for each brick individually